source/solvers/mf_navier_stokes.cc

```diff
@@ -264,23 +276,705 @@ MFNavierStokesSolver<dim>::assemble_system_rhs()
   this->system_rhs *= -1.0;
 }
 
-template <int dim>
+template <int dim, typename OperatorType>
 void
-MFNavierStokesSolver<dim>::update_multiphysics_time_average_solution()
+MFNavierStokesSolver<dim,
+                     OperatorType>::update_multiphysics_time_average_solution()
 {
   // TODO
 }
 
-template <int dim>
+template <int dim, typename OperatorType>
+double
+MFNavierStokesSolver<dim, OperatorType>::estimate_omega(
+  std::shared_ptr<OperatorType> &mg_operator)
+{
+  double omega = 0.0;
+
+  using SmootherPreconditionerType = DiagonalMatrix<VectorType>;
+  using ChebyshevPreconditionerType =
+    PreconditionChebyshev<OperatorType, VectorType, SmootherPreconditionerType>;
+  typename ChebyshevPreconditionerType::AdditionalData
+    chebyshev_additional_data;
+
+  chebyshev_additional_data.preconditioner =
+    std::make_shared<SmootherPreconditionerType>();
+  mg_operator->compute_inverse_diagonal(
+    chebyshev_additional_data.preconditioner->get_vector());
+  chebyshev_additional_data.constraints.copy_from(this->zero_constraints);
+  chebyshev_additional_data.degree               = 3;
+  chebyshev_additional_data.smoothing_range      = 20;
+  chebyshev_additional_data.eig_cg_n_iterations  = 40;
+  chebyshev_additional_data.eigenvalue_algorithm = ChebyshevPreconditionerType::
+    AdditionalData::EigenvalueAlgorithm::power_iteration;
+  chebyshev_additional_data.polynomial_type =
+    ChebyshevPreconditionerType::AdditionalData::PolynomialType::fourth_kind;
```

Comment:
Not all parameters need to be set here.

Reply:
That's true. We haven't experimented much with this yet. I think I will leave it like this in this PR, but later we will check this again.